Lib/idlelib/configdialog.py

```diff
@@ -1764,9 +1766,19 @@ class GenPage(Frame):
 
     def __init__(self, master):
         super().__init__(master)
+
+        self.init_validators()
         self.create_page_general()
         self.load_general_cfg()
 
+    def init_validators(self):
+        digits_or_empty_re = re.compile(r'[0-9]*')
+        def is_digits_or_empty(s):
+            "Return 's is blank or contains only digits'"
+            return digits_or_empty_re.fullmatch(s) is not None
```

Comment:
Possible immediate alternative:
```
try:
    return 0 <= int(s)
except ValueError:
     return False
```
Or, adding range check,
```
def rangecheck(widget, lo, hi):  # At global level, so can use on other tabs (and test).
    def inner(s):
        try:
            valid = lo <= int(s) <= hi
        except ValueError:
            valid = False
        if not valid: <showmessage: f"Entry must be int from {lo} to {hi}, inclusive">
        return valid
    return widget.register(inner, '%P')
# Example use.
        mincheck = rangecheck(frame_win_size, 10, 999)
        ...
                   validate='key', validatecommand=mincheck,
```

Reply:
> I mentioned not restricting validation to the General tab because I want to replace the indent slider with an Entry box. I would like to eventually move this to the first section of the General tab, but perhaps only when splitting that tab into two.

I'd like to leave that for later, then.

> Possible immediate alternative:
> 
> ```
> try:
>     return 0 <= int(s)
> except ValueError:
>      return False
> ```

I avoided this because `int()` allows leading '+' and '-' signs, which IMO we should disallow to keep things simple. (For example, `-0` is a valid non-negative integer.)

> It would be nice to check that an entry is in an allowed range, with an error message.

At a quick glance, the existing integer config options should either be non-negative, such as timeouts, or positive (i.e. non-zero), such as window width or minimum context lines. Following is a list with my suggested allowed ranges; let me know what you think, @terryjreedy.

1. Initial Window Width & Height: Positive
2. Completions Popup Wait (milliseconds): Non-negative
3. Paren-Match Time Displayed (zero = until next input): Non-negative
4. Format Paragraph Max. Width: Positive
5. Max. Context Lines: Positive
6. Auto-Squeeze Min. Lines: Non-negative (zero would mean the same as one, but allowing entering zero to have no minimum is harmless and could be intuitive to some)

Reply:
For a simple check, use the current code (with name change).
For a range check, check 1st char first, then int range.

1. Windows resize min is 10 (Command Prompt) or more -- enough to show '--  [_]  X' on the title bar.  A setting of 2 gives a window 10.x 11 point chars wide.  10-999 seem sensible to me for 1 and 4.

6. Someone requested (Stackoverflow, I think) 0 = infinity (large number).  This is similar to 3, which has explicit note about its convention.